openedx/core/djangoapps/content_libraries/rest_api/serializers.py

```diff
@@ -426,3 +434,101 @@ class LibraryBackupTaskStatusSerializer(serializers.Serializer):
     """
     state = serializers.CharField()
     url = serializers.FileField(source='file', allow_null=True, use_url=True)
+
+
+class LibraryRestoreFileSerializer(serializers.Serializer):
+    """
+    Serializer for restoring a library from a backup file.
+    """
+    # input only fields
+    file = serializers.FileField(write_only=True, help_text="A ZIP file containing a library backup.")
+
+    # output only fields
+    task_id = serializers.UUIDField(read_only=True)
+
+    def validate_file(self, value):
+        """
+        Validate that the uploaded file is a ZIP file.
+        """
+        if value.content_type != 'application/zip':
+            raise serializers.ValidationError("Only ZIP files are allowed.")
+        return value
+
+
+class LibraryRestoreTaskRequestSerializer(serializers.Serializer):
+    """
+    Serializer for requesting the status of a library restore task.
+    """
+    task_id = serializers.UUIDField(write_only=True, help_text="The ID of the restore task to check.")
+
+
+class RestoreSuccessDataSerializer(serializers.Serializer):
+    """
+    Serializer for the data returned upon successful restoration of a library.
+    """
+    learning_package_id = serializers.IntegerField(source="lp_restored_data.id")
+    title = serializers.CharField(source="lp_restored_data.title")
+    org = serializers.CharField(source="lp_restored_data.archive_org_key")
+    slug = serializers.CharField(source="lp_restored_data.archive_slug")
+    key = serializers.CharField(source="lp_restored_data.key")
+    archive_key = serializers.CharField(source="lp_restored_data.archive_lp_key")
+
+    containers = serializers.IntegerField(source="lp_restored_data.num_containers")
+    components = serializers.IntegerField(source="lp_restored_data.num_components")
+    collections = serializers.IntegerField(source="lp_restored_data.num_collections")
+    sections = serializers.IntegerField(source="lp_restored_data.num_sections")
+    subsections = serializers.IntegerField(source="lp_restored_data.num_subsections")
+    units = serializers.IntegerField(source="lp_restored_data.num_units")
+
+    created_on_server = serializers.CharField(source="backup_metadata.original_server", required=False)
+    created_at = serializers.DateTimeField(source="backup_metadata.created_at", format=DATETIME_FORMAT)
+    created_by = serializers.SerializerMethodField()
+
+    def get_created_by(self, obj):
```

Comment:
Please add in the docstring how this information is from the archive, and does not necessarily map to users in the system this view is being served from. I strongly suspect that at some point, someone will get confused between the two.